lib/playlistGenerator/PlaylistGenerator.js

```diff
@@ -267,7 +275,8 @@ var doPersist =  function() {
 };
 
 // save playlist
-var persistPlayList = function persistPlayList(force){
+// todo: if stream ended set presentationEndTime to now - playWindow
+var persistPlayList = function persistPlayList(force, ended = false){
```

Comment:
code changed. the playlist.json is read and then updated and written to disk directly. Changes in PlaylistGeneration reverted.